statistics-rest/src/main/java/org/uniprot/api/statistics/service/StatisticService.java

```diff
@@ -0,0 +1,85 @@
+package org.uniprot.api.statistics.service;
+
+import org.springframework.stereotype.Service;
+import org.uniprot.api.statistics.entity.UniprotkbStatisticsEntry;
+import org.uniprot.api.statistics.mapper.StatisticMapper;
+import org.uniprot.api.statistics.model.StatisticModuleStatisticCategory;
+import org.uniprot.api.statistics.model.StatisticModuleStatisticCategoryImpl;
+import org.uniprot.api.statistics.model.StatisticModuleStatisticType;
+import org.uniprot.api.statistics.repository.StatisticsCategoryRepository;
+import org.uniprot.api.statistics.repository.UniprotkbStatisticsEntryRepository;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class StatisticService {
+
+    private final UniprotkbStatisticsEntryRepository statisticsEntryRepository;
+    private final StatisticsCategoryRepository statisticsCategoryRepository;
+    private final StatisticMapper statisticMapper;
+
+    public StatisticService(
+            UniprotkbStatisticsEntryRepository uniprotkbStatisticsEntryRepository,
+            StatisticsCategoryRepository statisticsCategoryRepository,
+            StatisticMapper statisticMapper) {
+        this.statisticsEntryRepository = uniprotkbStatisticsEntryRepository;
+        this.statisticsCategoryRepository = statisticsCategoryRepository;
+        this.statisticMapper = statisticMapper;
+    }
+
+    public Collection<StatisticModuleStatisticCategory> findAllByVersionAndStatisticTypeAndCategoryIn(
+            String version, String statisticType, Collection<String> categories) {
```

Comment:
Please use specific interface like List unless your response can be List or Set then you can use Collection